common/datasets/librispeech.py

```diff
@@ -373,6 +375,34 @@ def get_bliss_lexicon(subdir_prefix=""):
     return merge_lexicon_job.out_bliss_lexicon
 
 
+def get_g2p_augmented_bliss_lexicon_dict(subdir_prefix=""):
+    """
+    Given the original LibriSpeech bliss lexicon, it is possible to estimate the pronunciation for
+    out of vocabulary (OOV) words for each of the LibriSpeech training corpora. Here, we create a dictionary
+    that has different train corpora as keys and the corresponding g2p augmented bliss lexicon as values
+
+    :param str subdir_prefix:
+    :return: dictionary of Paths to augmented bliss_lexicon
+    :rtype: dict[str, Path]
+    """
+    alias_path = os.path.join(subdir_prefix, "LibriSpeech", "lexicon")
+    augmented_bliss_lexica = {}
+
+    original_bliss_lexicon = get_bliss_lexicon(subdir_prefix=subdir_prefix)
+    g2p_augmenter = G2PBasedOovAugmenter(original_bliss_lexicon=original_bliss_lexicon)
+
+    bliss_corpus_dict = get_bliss_corpus_dict(subdir_prefix=subdir_prefix)
+    for corpus_name, bliss_corpus in bliss_corpus_dict.items():
+        if "train" in corpus_name:
+            augmented_bliss_lexica[corpus_name] = g2p_augmenter.get_g2p_augmented_bliss_lexicon(
+                bliss_corpus=bliss_corpus,
+                corpus_name=corpus_name,
+                alias_path=alias_path
```

Comment:
I would still suggest that we use the initial lexicon to train the g2p model and then extend it by the train-clean-100 OOV.
Then to create the train-clean-360-lexicon, we should augment the train-clean-100-lexicon by the remaining OOVs.

Not only do we save a little bit of computational effort on shared OOVs, but also would we be able to use the 100h alignment as input for 960h training jobs. (e.g. Pipeline: monophon -> 100h, triphon->460h, triphon->960h)
The only downside would be, that if we wanted to to 360h-only training, then a few unnecessary 100h OOV words would be added.

Reply:
> The only downside would be, that if we wanted to to 360h-only training, then a few unnecessary 100h OOV words would be added.

But this could be separated, so the 360h lexicon is "standalone" but the 460h lexicon is the 100h lexicon and then the additional OOVs from 360 added.

Reply:
> I would still suggest 

Please do a separate PR for this.